packages/core/src/app/payment/paymentMethod/CreditCardPaymentMethod.spec.tsx

```diff
@@ -196,16 +196,29 @@ describe('CreditCardPaymentMethod', () => {
         expect(defaultProps.deinitializePayment).toHaveBeenCalled();
     });
 
-    it('renders loading overlay while waiting for method to initialize', () => {
+    it('renders loading overlay while waiting for method to initialize', async () => {
         let component: ReactWrapper;
+        const mockInitializePayment = jest.fn().mockImplementation(() => {
+            return new Promise((resolve) => resolve(true));
+        });
 
         component = mount(<CreditCardPaymentMethodTest {...defaultProps} isInitializing />);
 
         expect(component.find(LoadingOverlay).prop('isLoading')).toBe(true);
 
-        component = mount(<CreditCardPaymentMethodTest {...defaultProps} />);
+        component = mount(
+            <CreditCardPaymentMethodTest
+                {...defaultProps}
+                initializePayment={mockInitializePayment}
+                isInitializing={false}
+            />
+        );
+
+        const loadingOverlay = component.find(LoadingOverlay);
+        setTimeout(() => {
```

Comment:
Did you try how it will be working with `process.nextTick()`?

Reply:
It did not work, the problem is we need to wait until initializePayment finish and then check loading overlay, but in test we have updated isLoading prop after we get LoadingOverlay component because of initializePayment async. So i've used setTimeout to rearrange call stack